A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/EICAS_Common.js

```diff
@@ -79,16 +79,22 @@ class EICASCommonDisplay extends Airliners.EICASTemplateElement {
     refreshGrossWeight(_force = false) {
         let gw = 0;
         let isInMetric = BaseAirliners.unitIsMetric(Aircraft.A320_NEO);
+        let unit;
         if (isInMetric) {
-            gw = Math.round(SimVar.GetSimVarValue("TOTAL WEIGHT", "kg"));
-            if (this.gwUnit)
+            if (this.gwUnit) {
```

Comment:
Ok, let me check and refactor them. Thanks for you help.

Reply:
@UncleClapton can we move the display unit checking process to init() ? As we just need it to set it only one time ?

Reply:
This code probably exists because unit settings may be introduced in the future. Probably best to leave it in the update loop for now until that's clarified down the road.

Reply:
@UncleClapton  I fixed and refactored the code base on your suggestion.